addons/purchase_stock/models/purchase.py

```diff
@@ -448,7 +448,9 @@ def _create_or_update_picking(self):
                 # If the user increased quantity of existing line or created a new line
                 pickings = line.order_id.picking_ids.filtered(lambda x: x.state not in ('done', 'cancel') and x.location_dest_id.usage in ('internal', 'transit', 'customer'))
                 picking = pickings and pickings[0] or False
-                if not picking and line.product_qty > line.qty_received:
+                if not picking:
+                    if not line.product_qty > line.qty_received:
+                        return
```

Comment:
`continue` if you have other lines in `self`

Reply:
Nice catch! Thank you for the review!